lib/segment/src/segment_constructor/segment_builder.rs

```diff
@@ -552,13 +552,6 @@ impl SegmentBuilder {
             #[cfg(not(feature = "gpu"))]
             let gpu_device = None;
 
-            // If GPU is enabled, release all CPU cores except one.
-            if let Some(_gpu_device) = &gpu_device {
-                if permit.num_cpus > 1 {
-                    permit.release_cpu_count(permit.num_cpus - 1);
-                }
-            }
```

Comment:
This behavior was broken and would have created problems with changes of this PR. E.g. we don't want to schedule CPU optimization if GPU is available. More correct implementation should involve dedicated GPU permit.